scrapy/commands/parse.py

```diff
@@ -205,12 +205,14 @@ def callback(response):
                     req.meta['_depth'] = depth + 1
                     req.meta['_callback'] = req.callback
                     req.callback = callback
+                    req.callback = original_callback
```

Comment:
Hi @nctl144, thanks for the PR! :)

At this point, the custom callback method was not called yet. You are simply resetting the requests callback to its original _immediately_ after it had been set to the custom callback. Thereby, you disable the whole functionality of `scrapy parse` for all subsequent requests, and instead of printing the requests and items of the _last_ request, scrapy will print those of the first request.

You can see this if you yield an item before printing "done", like this:
```python
import scrapy


class HttpBinSpider(scrapy.Spider):

    name = "httpbin.org"

    start_urls = ['https://httpbin.org/']

    def parse(self, response):
        if response.meta.get('retried'):
            self.logger.info("done")
            yield {'some': 'item'}
            return
        response.meta['retried'] = True
        yield response.request.replace(dont_filter=True)
```

With this PR, the end of the crawl log with `scrapy parse "https://httpbin.org/" -d 2` looks like this:
```
>>> STATUS DEPTH LEVEL 1 <<<
# Scraped Items  ------------------------------------------------------------
[]
# Requests  -----------------------------------------------------------------
[<GET https://httpbin.org/>]
```

whereas with a correct fix it should look like this:
```
>>> STATUS DEPTH LEVEL 2 <<<
# Scraped Items  ------------------------------------------------------------
[{'some': 'item'}]
# Requests  -----------------------------------------------------------------
[]
```

The real call to the original callback is done in [line 195](https://github.com/scrapy/scrapy/blob/master/scrapy/commands/parse.py#L195), so you should try to restore the original `request.callback` attribute somewhere around there.